service/dap/server.go

```diff
@@ -1221,7 +1233,7 @@ func (s *Server) convertVariableWithOpts(v *proc.Variable, qualifiedNameOrExpr s
 				// frame-independent address-based expression, not fully-qualified name.
 				loadExpr := fmt.Sprintf("*(*%q)(%#x)", typeName, v.Addr)
 				s.log.Debugf("loading %s (type %s) with %s", qualifiedNameOrExpr, typeName, loadExpr)
-				vLoaded, err := s.debugger.EvalVariableInScope(-1, 0, 0, loadExpr, DefaultLoadConfig)
+				vLoaded, err := s.debugger.EvalVariableInScope(-1, 0, 0, loadExpr, loadConfig)
```

Comment:
It's interesting that the auto-loading limits change depending on whether you are expanding a Variable from Variables pane or from expression. This deserve a test case. 

I am also wondering now if we should treat all variable expansions this way. If a user manually tries to expand a variable, then that's a single human action as well. At the same time, if the user leaves a field expanded and steps, it stay expanded and will be automatically reloaded, so maybe this is more like a watch and we should be conservative.

Reply:
Now narrowed the scope where this heuristic applies further and this applies only to the string type, and only from the evaluate request handler. So, convertVariableWithOpts isn't affected.

I'd go safe and stay with short limit now. I think, eventually users have to look into the data further through the repl mode. Allowing to sett different limits for the repl mode and fully supporting the dlv cli's expression syntaxes, and then ask users to use dlv expression to look into further from there may be more helpful and good for users for long term..

Reply:
I learned that expanded variables remain open and kept retrieved. In that case, loading all automatically may not be ideal. We need more careful look on the behavior of Variables section when implementing autoloading of strings and slices/arrays/large structs.

Reply:
I agree with narrowing the scope, starting small and considering expansions based on user feedback. I too was worried about the impact of reloading due to sticky expansion (see comment from Apr 6). Btw, we already load large structs completely because MaxStructFields defaults to -1.